app/models/service_deployment.rb

```diff
@@ -50,4 +48,11 @@ def fail!(retryable: false)
   def pending?
     [STATUS[:queued], STATUS[:deploying]].include?(status)
   end
+
+  def self.generate_github_link(service:, commit_sha:)
+    url_link = Service.find(service.id).git_repo_url
```

Comment:
If you're passing in the service, you don't need to find the service again
But also, if you made this an instance method rather than a class method, then the instance would already have service &  commit_sha available as instance variables, so you wouldn't need to pass any arguments at all